src/panels/lovelace/cards/energy/hui-energy-devices-detail-graph-card.ts

```diff
@@ -148,6 +148,24 @@ export class HuiEnergyDevicesDetailGraphCard
       { num: formatNumber(total, this.hass.locale), unit: UNIT }
     );
 
+  private _nameToStatId(name: string): string | undefined {
+    return this._data?.prefs.device_consumption.find(
+      (d) =>
+        (d.name ??
+          getStatisticLabel(
+            this.hass,
+            d.stat_consumption,
+            this._data?.statsMetadata[d.stat_consumption]
+          )) === name
+    )?.stat_consumption;
+  }
```

Comment:
I notice with echarts change we switched from tracking hidden datasets by id to tracking them by display name. 
This makes me a little nervous that this will fall apart if any two entities have same display name. This change needs to know the hidden ids to compute some data, so I try to extract the id back from the name, but perhaps that needs some re-thought. Maybe that's just an edge case.